src/components/auth/Login.js

```diff
@@ -0,0 +1,185 @@
+import React, { useState, useEffect } from 'react';
+import PropTypes from 'prop-types';
+import { connect } from 'react-redux';
+import Avatar from '@material-ui/core/Avatar';
+import Button from '@material-ui/core/Button';
+import CssBaseline from '@material-ui/core/CssBaseline';
+import TextField from '@material-ui/core/TextField';
+import FormControlLabel from '@material-ui/core/FormControlLabel';
+import Checkbox from '@material-ui/core/Checkbox';
+import Link from '@material-ui/core/Link';
+import Grid from '@material-ui/core/Grid';
+import Box from '@material-ui/core/Box';
+import LockOutlinedIcon from '@material-ui/icons/LockOutlined';
+import Typography from '@material-ui/core/Typography';
+import { makeStyles } from '@material-ui/core/styles';
+import Container from '@material-ui/core/Container';
+
+import { setAlert } from '../../actions/alertAction';
+import { login, clearErrors } from '../../actions/authAction';
+
+const Copyright = () => {
+  return (
+    <Typography variant='body2' color='textSecondary' align='center'>
+      {'Copyright © '}
+      <Link color='inherit' href='https://material-ui.com/'>
+        Your Website
+      </Link>{' '}
+      {new Date().getFullYear()}
+      {'. Built with '}
+      <Link color='inherit' href='https://material-ui.com/'>
+        Material-UI.
+      </Link>
+    </Typography>
+  );
+};
+
+const useStyles = makeStyles(theme => ({
+  '@global': {
+    body: {
+      backgroundColor: theme.palette.common.white
+    }
+  },
+  paper: {
+    marginTop: theme.spacing(8),
+    display: 'flex',
+    flexDirection: 'column',
+    alignItems: 'center'
+  },
+  avatar: {
+    margin: theme.spacing(1),
+    backgroundColor: theme.palette.secondary.main
+  },
+  form: {
+    width: '100%', // Fix IE 11 issue.
+    marginTop: theme.spacing(1)
+  },
+  submit: {
+    margin: theme.spacing(3, 0, 2)
+  }
+}));
+
+const Login = ({
+  setAlert,
+  auth: { error, isAuthenticated },
+  login,
+  clearErrors
+}) => {
+  const [user, setUser] = useState({
+    email: '',
+    password: ''
+  });
+
+  useEffect(() => {
+    if (isAuthenticated) {
+      // props.history.push('/')?
+      window.location = '/';
```

Comment:
Yes! This is the way to do the routing lol But you need `history` from `Router` to be passed as `props` to this component in order for `Login` component to have access to `props.history`.